File: ToolDemo/app/src/main/java/com/lilanz/tooldemo/utils/WebViewUtil.java
package com.lilanz.tooldemo.utils;

import android.content.Context;
import android.webkit.WebSettings;
import android.webkit.WebView;

public class WebViewUtil {

    /**
     * 初始化webView
     * @param context
     * @param webView
     */
    public static void init(Context context, WebView webView){
        String appCachePath = context.getCacheDir().getAbsolutePath();
        WebSettings webSettings = webView.getSettings();
        webSettings.setAllowUniversalAccessFromFileURLs(true);
        webSettings.setAllowFileAccess(true);
        webSettings.setAllowFileAccessFromFileURLs(true);
        //用于js调用Android
        webSettings.setJavaScriptEnabled(true);
        //设置编码方式
        webSettings.setDefaultTextEncodingName("utf-8");
        webSettings.setDomStorageEnabled(true);
        webSettings.setAppCacheMaxSize(1024 * 1024 * 8);
        webSettings.setAppCachePath(appCachePath);

        webSettings.setAllowFileAccess(true);
        webSettings.setAppCacheEnabled(true);

        webSettings.setAllowContentAccess(true);
        webSettings.setBuiltInZoomControls(true);
        webSettings.setDisplayZoomControls(false);

        webSettings.setLoadWithOverviewMode(true);
        webSettings.setUseWideViewPort(true);
        webSettings.setGeolocationEnabled(true);
        webSettings.setDatabaseEnabled(true);
        webSettings.setSupportZoom(true);
    }
}

File: KotlinTool/app/src/main/java/com/lilanz/kotlintool/utils/WebViewUtil.kt
package com.lilanz.kotlintool.utils

import android.content.Context
import android.webkit.WebView

class WebViewUtil {
    companion object {
        /**
         * 初始化webview
         */
        fun init(context: Context, webView: WebView) {
            val appCachePath = context.cacheDir.absolutePath
            val webSettings = webView.settings
            webSettings.allowUniversalAccessFromFileURLs = true
            webSettings.allowFileAccess = true
            webSettings.allowFileAccessFromFileURLs = true
            //用于js调用Android
            webSettings.javaScriptEnabled = true
            //设置编码方式
            webSettings.defaultTextEncodingName = "utf-8"
            webSettings.domStorageEnabled = true
            webSettings.setAppCacheMaxSize(1024 * 1024 * 8.toLong())
            webSettings.setAppCachePath(appCachePath)

            webSettings.allowFileAccess = true
            webSettings.setAppCacheEnabled(true)

            webSettings.allowContentAccess = true
            webSettings.builtInZoomControls = true
            webSettings.displayZoomControls = false

            webSettings.loadWithOverviewMode = true
            webSettings.useWideViewPort = true
            webSettings.setGeolocationEnabled(true)
            webSettings.databaseEnabled = true
            webSettings.setSupportZoom(true)
        }
    }
}